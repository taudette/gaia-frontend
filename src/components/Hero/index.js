import React from 'react';
import styles from '../../styles/components/_hero.scss'

const HeroComponent = (props) => {
  return (
    <section >
      <div className={styles.heroContainer}>
        <img src={props.hero.lgImage} alt="hero-image" />
        <div className={styles.text}>
          <h2>{props.hero.name}</h2>
          <p className="hidden-xs"><small>{props.hero.body}</small></p>
        </div>
      </div>
    </section>
  )
}

export default HeroComponent


