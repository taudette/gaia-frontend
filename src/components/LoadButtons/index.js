import React from 'react';
import styles from '../../styles/components/_loadButtons.scss'
import { Row, Col } from 'react-bootstrap'

const LoadButtonsComponent = (props) => {
  return (
    <Row className={styles.buttonRow}>
      {/* don't show load more button if no more to load*/}
      { props.endIndex != props.listLength &&
        <Col smOffset={4} sm={4} xs={12} className="text-center">
          <button className={styles.loadButton} onClick={props.addVideos}>Load More</button>
        </Col>
      }
      <Col sm={4} xs={12} className="text-right">
        <button onClick={props.goToTop} className={styles.upButton}><i className={styles.arrow}></i></button>
      </Col>
    </Row>
  )
}

export default LoadButtonsComponent