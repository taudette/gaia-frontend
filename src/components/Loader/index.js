import React from 'react';
import loader from '../../styles/components/_loader.scss';

const LoaderComponent = () => (
  <div className={loader.loaderContainer}>
    <div className={loader.loader}></div>
  </div>
)

export default LoaderComponent