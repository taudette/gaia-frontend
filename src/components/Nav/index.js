import React from 'react';

export const NavComponent = () => (
  <div>
    Nav Component
  </div>
)

export default NavComponent
