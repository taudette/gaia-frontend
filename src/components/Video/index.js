import React from 'react';
import styles from '../../styles/components/_video.scss'

export const VideoComponent = (props) => {
  console.log(props)
  return (
      <div>
        {/* only render video if image and title are avaiable, check the rest individually */}
        { props.img && props.title && 
          <div className={styles.tileContainer}>
            <img src={props.img} alt="video-image" /> 
            <p className="">{props.title}</p> 
            { props.segment && <p>segment: {props.segment}</p> }
            { props.likes && <p>likes: {props.likes}</p> }
            { props.dislikes && <p>dislikes: {props.dislikes}</p> }
            { props.seasons && <p>seasons: {props.seasons} </p> } 
            { props.episodes && <p>episodes: {props.episodes} </p> } 
          </div>
        }
      </div>
      
  )
}

export default VideoComponent
