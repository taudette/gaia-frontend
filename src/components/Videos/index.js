import React from 'react';
import VideoComponent from '../Video'
import styles from '../../styles/components/_videos.scss';

export const VideosComponent = (props) => {
  if (props.loading) {
    return <p>Loading...</p>
  }

  return (
    <div >
      {props.videos &&
        <ul className={styles.videoList}>
          {props.videos.map((video, index) => {
            return <VideoComponent {...video} key={index} />
          })}
        </ul>
      }
    </div>
  )
}

export default VideosComponent
