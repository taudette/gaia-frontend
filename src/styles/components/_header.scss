@import '../base/_settings.scss';

header {
  padding-left: 4rem;
  padding-right: 4rem;
  display: flex;
  align-items: center;
  height: 5rem;
}

.row {
  width: 100%;
}

.logoContainer {
  width: 7rem;
}

.userContainer {
  float: right;
  text-align: right;

  img {
    width: 2rem;
    margin-left: 1rem;
  }

  span {
    color: darkgrey;
  }
}

.icon {
  background-image: url("../../../public/images/header-logo.svg");
}

