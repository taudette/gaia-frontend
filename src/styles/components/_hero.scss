img {
  width: 100%;
}

.heroContainer {
  position: relative;
  text-align: center;
  color: white;
  margin-bottom: 4rem;
}

.text {
  position: absolute;
  top: 50%;
  left: 40%;
  transform: translate(-50%, -50%);
  text-align: left;
    h2 {
      text-transform: uppercase;
    }
}