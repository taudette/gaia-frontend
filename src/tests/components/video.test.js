import React from 'react';
import { shallow } from 'enzyme';
import { VideoComponent } from '../../components/Video';
import video from '../fixtures/video';

test('should render video', () => {
  const wrapper = shallow(<VideoComponent {...video} key="1" />)
  expect(wrapper).toMatchSnapshot();
})

test('should render video component without video', () => {
  const wrapper = shallow(<VideoComponent  />)
  expect(wrapper).toMatchSnapshot();
})