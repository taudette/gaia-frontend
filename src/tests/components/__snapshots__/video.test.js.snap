// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render video 1`] = `
<Col
  bsClass="col"
  componentClass="div"
  lg={3}
  md={4}
  sm={6}
  xs={12}
>
  <li
    className="tile"
  >
    <div
      className="tileContainer"
    >
      <img
        alt="video-image"
        src="https://img.gtvcdn.com/cdn/farfuture/29uIix8RhExtaI54apLJdD1J62mat_Cxx6mMQNxxUek/mtime%3A1496214668/sites/default/files/imagecache/hero_320x200/img_16x9_landsacpe_title/142666_wt_s26e5_the-mystery-of-comet-shoemaker-levy-9_16x9.jpg"
      />
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          xs={5}
        >
          <a
            className="segment"
            href="javascript:void(0);"
          >
            Seeking Truth
          </a>
        </Col>
        <Col
          bsClass="col"
          className="text-right"
          componentClass="div"
          xs={7}
        >
          <a
            href="javascript:void(0);"
          >
            likes: 
            888
             
          </a>
          <a
            href="javascript:void(0);"
          >
            dislikes: 
            48
          </a>
        </Col>
      </Row>
      <div
        className="textContainer"
      >
        <Row
          bsClass="row"
          componentClass="div"
        >
          <a
            className="title"
            href="javascript:void(0);"
          >
            [#220] The Mystery of Comet Shoemaker-Levy 9
          </a>
        </Row>
        <Row
          bsClass="row"
          componentClass="div"
        />
      </div>
    </div>
  </li>
</Col>
`;

exports[`should render video component without video 1`] = `
<Col
  bsClass="col"
  componentClass="div"
  lg={3}
  md={4}
  sm={6}
  xs={12}
>
  <li
    className="tile"
  />
</Col>
`;
